app/src/main/java/com/mk/tasky/agenda/data/AgendaRepositoryImpl.kt

```diff
@@ -113,9 +127,48 @@ class AgendaRepositoryImpl(
         }
     }
 
-    private suspend fun deleteReminderByIdRemotely(id: String): Result<Unit> {
+    override suspend fun insertTask(task: Task, isEdit: Boolean): Result<Unit> {
+        dao.insertTask(task.toEntity())
+        return saveTaskRemotely(task, isEdit)
+    }
+
+    override suspend fun changeStatusTask(id: String, isDone: Boolean): Result<Unit> {
+        val task = getTaskById(id).copy(
+            isDone = isDone
+        )
+        return insertTask(task, true)
```

Comment:
Just a recommendation, feel free to ignore: I'm a fan of always using named parameters to make things easier readable. The IDE will give you hints, but there's no way a code reviewer could tell what this `true` stands for (without looking at the function signature)